ServerCore/Pages/Events/AddAdminOrAuthor.cshtml.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Events
+{
+    [Authorize("IsEventAdmin")]
+    public class AddAdminOrAuthorModel : EventSpecificPageModel
+    {
+        public IList<PuzzleUser> Users { get; set; }
+
+        public bool addAdmin { get; set; }
+
+        public AddAdminOrAuthorModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        public async Task<IActionResult> OnGetAsync(bool addAdmin)
+        {
+            this.addAdmin = addAdmin;
+
+            if (addAdmin)
+            {
+                Users = await (from user in _context.PuzzleUsers
```

Comment:
This is all users of all events of all time. Do we have any reasonable way to narrow it down?

Reply:
Added an issue for that type of functionality because yes: https://github.com/PuzzleServer/mainpuzzleserver/issues/220